tests/Integration/inc/Engine/Optimization/RUCSS/Admin/Subscriber/deleteUsedCssOnUpdateOrDelete.php

```diff
@@ -0,0 +1,77 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Tests\Integration\inc\Engine\Optimization\RUCSS\Admin\Subscriber;
+
+use WP_Rocket\Tests\Integration\DBTrait;
+use WP_Rocket\Tests\Integration\TestCase;
+
+/**
+ * @covers \WP_Rocket\Engine\Optimization\RUCSS\Admin\Subscriber::delete_used_css_on_update_or_delete
+ *
+ * @group  RUCSS
+ */
+class Test_DeleteUsedCssOnUpdateOrDelete extends TestCase{
+	use DBTrait;
+
+	private $posts;
+	private $input;
+
+	public static function setUpBeforeClass(): void {
+		self::installFresh();
+
+		parent::setUpBeforeClass();
+	}
+
+	public static function tearDownAfterClass() {
+		parent::tearDownAfterClass();
+
+		self::uninstallAll();
+	}
+
+	public function tearDown() : void {
+		remove_filter( 'pre_get_rocket_option_remove_unused_css', [ $this, 'set_rucss_option' ] );
+
+		parent::tearDown();
+	}
+
+	/**
+	 * @dataProvider configTestData
+	 */
+	public function testShouldTruncateTableWhenOptionIsEnabled( $input ){
+		$container           = apply_filters( 'rocket_container', null );
+		$rucss_usedcss_query = $container->get( 'rucss_used_css_query' );
+
+		$this->input = $input;
+		add_filter( 'pre_get_rocket_option_remove_unused_css', [ $this, 'set_rucss_option' ] );
+
+		foreach ( $input['items'] as $item ) {
+			$post          = $this->factory->post->create_and_get();
+			$url           = untrailingslashit( get_permalink( $post->ID ) );
+			$item['url']   = $url;
+			$this->posts[] = $post;
+
+			$rucss_usedcss_query->add_item( $item );
+		}
+
+		$result = $rucss_usedcss_query->query();
+		$this->assertCount( count( $input['items'] ), $result );
+
+		foreach ( $this->posts as $post ) {
+			do_action( 'delete_post', $post->ID );
+		}
+
+		$rucss_usedcss_query = $container->get( 'rucss_used_css_query' );
```

Comment:
Also here why defining it twice?

Reply:
All these tests are used for deleting from DB, right?

So the idea behind each test is simple:
1. Insert into the DB
2. Count the number of inserted values into DB
3. Clean the DB
4. Count again the left values in the DB

Based on each test: delete which are older than 1 month, truncate table, delete on update, the expectations are setup.
eg. for truncate => always the select after truncate should match 0 items in the DB
eg. for older than 1 month => I insert 2 items, 1 is not old and 1 is older than 1 month. And the expectation is that : After insert I have 2 rows in the DB. After cleaning the old files I have only 1  row in the DB.